### ISSUE TITLE
Pupil Player with Python3.5 - offline pupil detection memory consumption --> crash

### ISSUE THREAD

**Issue by willpatera** [pupil-labs/pupil#889]
## Overview
Pupil Player crashes during offline pupil detection from current commit in master branch - https://github.com/pupil-labs/pupil/commit/dc5f1543576a24ead75dd757ceac51d9ce4005e5. This is the first time I have seen this behavior. The behavior is easily reproducible on my development machine. The system monitor shows system memory usage of 100% at the time of the crash (which appears to be consistent with the error logged below).

## System Info
OS: Ubuntu 16.04 LTS (64 bit)
Memory: 7.7 GiB (8gb)
Processor: Intel® Core™ i5-6260U CPU @ 1.80GHz × 4 
Graphics: Intel® Iris Graphics 540 (Skylake GT3e) 
Python: Using python3.5.2 (system python3 on Ubuntu 16.04 LTS)

## Log
```bash
player - [ERROR] launchables.player: Process Player crashed with trace:
Traceback (most recent call last):
  File "/home/pupil-labs/Documents/pupil_labs/pupil/pupil_src/launchables/player.py", line 441, in player
    p.recent_events(events)
  File "/home/pupil-labs/Documents/pupil_labs/pupil/pupil_src/shared_modules/gaze_producers.py", line 389, in recent_events
    self.calibrate_section(s)
  File "/home/pupil-labs/Documents/pupil_labs/pupil/pupil_src/shared_modules/gaze_producers.py", line 449, in calibrate_section
    sec['bg_task'] = bh.Task_Proxy('{}'.format(self.sections.index(sec) + 1), calibrate_and_map, args=generator_args)
  File "/home/pupil-labs/Documents/pupil_labs/pupil/pupil_src/shared_modules/background_helper.py", line 35, in __init__
    self.process.start()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 212, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/context.py", line 267, in _Popen
    return Popen(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 20, in __init__
    self._launch(process_obj)
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 67, in _launch
    self.pid = os.fork()
OSError: [Errno 12] Cannot allocate memory

player - [INFO] launchables.player: Process shutting down.
```

**Comment by willpatera** [comment]
Note - this is specific to `python3.5` and should not be an issue if using `python3.6`

**Comment by papr** [comment]
This issue will be fixed by implementing an on-demand de-serialization. This will not be implemented for `v1.0` but in the next release.

**Comment by cpicanco** [comment]
@willpatera @fneitzel @papr #895.

I have some recordings with 20 min (3.8 gb, world, 5.8 gb eye) and they are taking forever to (not) open. Right now I couldn't open any of them. Some other recordings with 6.5 min (1.3 gb eye, 1.6 gb world) will open normally. 

OS: Ubuntu 16.04 LTS (64 bit)
Memory: 2.0 gb, ddr3, 1333
Processor: i3
Player Version: 0.9.14.7

**Comment by cpicanco** [comment]
Reproduced with 3d method. 2d is hanging a lot, but will complete.

```
Process player:
Traceback (most recent call last):
  File "multiprocessing/process.py", line 249, in _bootstrap
  File "multiprocessing/process.py", line 93, in run
  File "launchables/player.py", line 447, in player
  File "shared_modules/gaze_producers.py", line 243, in on_notify
  File "shared_modules/gaze_producers.py", line 333, in calibrate_section
  File "shared_modules/background_helper.py", line 35, in __init__
  File "multiprocessing/process.py", line 105, in start
  File "multiprocessing/context.py", line 223, in _Popen
  File "multiprocessing/context.py", line 277, in _Popen
  File "multiprocessing/popen_fork.py", line 20, in __init__
  File "multiprocessing/popen_fork.py", line 67, in _launch
OSError: [Errno 12] Cannot allocate memory
```

**Comment by sachabroute** [comment]
Hello,
I am using Pupil Player v1.1 and am currently experiencing this error while using the Offline Surface Tracker Plugin on ~7min recordings. It seems to happen while building the Marker Cache. When will this be fixed/is there a current workaround?

`player - [ERROR] launchables.player: Process Player crashed with trace:
Traceback (most recent call last):
  File "launchables/player.py", line 415, in player
  File "launchables/player.py", line 393, in handle_notifications
  File "shared_modules/plugin.py", line 318, in add
  File "shared_modules/offline_surface_tracker.py", line 65, in __init__
  File "shared_modules/offline_surface_tracker.py", line 293, in init_marker_cacher
  File "multiprocessing/process.py", line 105, in start
  File "multiprocessing/context.py", line 277, in _Popen
  File "multiprocessing/popen_fork.py", line 20, in __init__
  File "multiprocessing/popen_fork.py", line 67, in _launch
OSError: [Errno 12] Cannot allocate memory

player - [INFO] launchables.player: Process shutting down.
`
OS: Ubuntu 16.04 LTS (64 bit)
Memory: 7.7 GiB
Processor: Intel® Core™ i5-5200 CPU @ 2.20GHz × 4
Player Version: v1.1-2

**Comment by sachabroute** [comment]
Hello,

Adding to my comment above:
Using Ubuntu 17.10 (which has python3.6 instead of python3.5) an issue is still present which makes the computer freezes completely. I had to force shutdown the computer.
However, on Windows and Mac it seems to work correctly with my ~7min recordings. The marker cache was able to build -- So I could not reproduce the issue.

**Comment by papr** [comment]
This issue has been fixed with deferred deserialization in the v1.8 release.
